Sources/Quick/DSL/World+DSL.swift

```diff
@@ -20,8 +20,13 @@ extension World {
 
     internal func describe(description: String, flags: FilterFlags, closure: () -> ()) {
         guard currentExampleGroup != nil else {
-            NSException(name: NSInternalInconsistencyException, reason:"Error: example group was not created by its parent QuickSpec spec. Check that describe() or context() was used in QuickSpec.spec() and not a more general context (i.e. an XCTestCase test)", userInfo: nil).raise()
-            return
+            let error = "Error: example group was not created by its parent QuickSpec spec. Check that describe() or context() was used in QuickSpec.spec() and not a more general context (i.e. an XCTestCase test)"
+            #if os(Linux)
+                fatalError(error)
+            #else
+                NSException(name: NSInternalInconsistencyException, reason: error, userInfo: nil).raise()
+                return ()
+            #endif
```

Comment:
Some context on why we're using `NSException` here in the first place:

@morganchen12 was kind enough to include the very descriptive error message you see above in #431. When he did, I asked him to include a unit test for the change. Unfortunately, we can't test `fatalError` (as far as I can tell), so we stuck with `NSException` here, since we can test that an `NSException` is thrown.

We'd be able to get rid of these conditional macros and the `NSException` if we had a good plan for how to test that users actually see this nice error message.